libs/ml/include/ml/optimisation/optimiser.hpp

```diff
@@ -26,9 +26,6 @@
 #include "ml/optimisation/learning_rate_params.hpp"
```

Comment:
you need to move as many of these includes to source as possible. For example, mean.hpp can definitely be moved.